client/lib/application/src/access-form/access-form.jsx

```diff
@@ -82,15 +84,24 @@ class AccessForm extends React.Component {
         });
     }
 
+    fetchScopes(resourceType) {
+      this.props.essentialsActions.fetchScopes(resourceType).then(val => {
+        const state = Object.assign({}, this.state);
```

Comment:
No need to change it, but nicer would be:

```
const applicationScopes = this.state.applicationScopes.set(val[0], val[1].reduce((map, res) => map.set(res.id, Immutable.fromJS(res)), Immutable.Map()));
this.setState({applicationScopes});
```

Reply:
done